### ISSUE TITLE
Truncating class ignores super/sub-classes (leaves references in superclass indices, leaves subclass records in place)

### ISSUE THREAD

**Issue by kstock** [orientechnologies/orientdb#3919]
detected in 2.0.6
-----
If I have
an index on V.bid
and a class VV extends V
and I truncate VV

then there will exist dangling references to records from the VV class in the index, hence 
select from VV 
and 
select from VV  where bid = 1
can return different results.

I have to rebuild the index in order to fix things.

The expected behavior is to not have to rebuild the index, or to at least have a warning in the documentation. 

This issue is similar to 
(Truncate vertex class not delete dangled Edges)
https://github.com/orientechnologies/orientdb/issues/2280
in that it is about dangling references.

The subclass issue is that if I truncate V it will leave the VV records in place. I like this behavior but it is confusing because the console tells me it truncated the amount of vertices in V and all its subclasses (note below it says Truncated 5 record(s) ). Ideally you would be able to optionally truncate recursively and the documentation/console messages would reflect what was actually happening

Superclass indices example:

```
orientdb> create database plocal:/tmp/foobarbaz admin admin

Creating database [plocal:/tmp/foobarbaz] using the storage type [plocal]...
Database created successfully.

Current database is: plocal:/tmp/foobarbaz
orientdb {db=foobarbaz}> create property V.bid STRING

Property created successfully with id=1

orientdb {db=foobarbaz}> create index V.bid UNIQUE_HASH_INDEX


Creating index...
Created index successfully with 0 entries in 0.297000 sec(s).


Index created successfully
orientdb {db=foobarbaz}> create class VV extends V

Class created successfully. Total classes in database now: 11

orientdb {db=foobarbaz}> insert into VV set bid = 1

Inserted record 'VV#11:0{bid:1} v1' in 0.014000 sec(s).

orientdb {db=foobarbaz}> truncate class VV

Truncated 1 record(s) in 0.002000 sec(s).

orientdb {db=foobarbaz}> select from VV where bid = 1

----+-----+------+----
#   |@RID |@CLASS|bid
----+-----+------+----
0   |#11:0|VV    |1
----+-----+------+----

1 item(s) found. Query executed in 0.001 sec(s).
orientdb {db=foobarbaz}> select from VV


0 item(s) found. Query executed in 0.001 sec(s).
```

Truncate subclass example:
```
orientdb {db=foobarbaz}> select from V

----+-----+------+----
#   |@RID |@CLASS|foo
----+-----+------+----
0   |#11:0|VV    |1
1   |#11:1|VV    |1
2   |#11:2|VV    |1
3   |#11:3|VV    |1
4   |#11:4|VV    |1
----+-----+------+----

5 item(s) found. Query executed in 0.002 sec(s).
orientdb {db=foobarbaz}> truncate class V

Truncated 5 record(s) in 0.025000 sec(s).

orientdb {db=foobarbaz}> select from V

----+-----+------+----
#   |@RID |@CLASS|foo
----+-----+------+----
0   |#11:0|VV    |1
1   |#11:1|VV    |1
2   |#11:2|VV    |1
3   |#11:3|VV    |1
4   |#11:4|VV    |1
----+-----+------+----

5 item(s) found. Query executed in 0.001 sec(s).
```

**Comment by luigidellaquila** [comment]
I'm on it

**Comment by kstock** [comment]
I've noticed that "move vertex" has the same issue.

If I have 

V.myid  UNIQUE_HASH_INDEX
MYV extends V

and I try to "move vertex" a V vertex to MYV 

 MOVE VERTEX #17:44791 TO CLASS:MYV

It  errors with something like:

com.orientechnologies.orient.core.storage.ORecordDuplicatedException: Cannot index record MYV{myid:78957,...}: found duplicated key '78957' in index 'V.ovgid' previously assigned to the record #17:44791 RID=#17:44791

**Comment by luigidellaquila** [comment]
Fixed in "develop" branch, it will be released in 2.1 rc2
